python/pyarrow/tests/test_parquet.py

```diff
@@ -577,31 +577,42 @@ def test_parquet_metadata_api():
 
 @parquet
 @pytest.mark.parametrize(
-    'data, dtype, min_value, max_value, null_count, num_values',
+    (
+        'data',
+        'dtype',
+        'physical_type',
+        'min_value',
+        'max_value',
+        'null_count',
+        'num_values',
+        'distinct_count'
+    ),
     [
-        ([1, 2, 2, None, 4], np.uint8, 1, 4, 1, 4),
-        ([1, 2, 2, None, 4], np.uint16, 1, 4, 1, 4),
-        ([1, 2, 2, None, 4], np.uint32, 1, 4, 1, 4),
-        ([1, 2, 2, None, 4], np.uint64, 1, 4, 1, 4),
-        ([-1, 2, 2, None, 4], np.int16, -1, 4, 1, 4),
-        ([-1, 2, 2, None, 4], np.int32, -1, 4, 1, 4),
-        ([-1, 2, 2, None, 4], np.int64, -1, 4, 1, 4),
-        ([-1.1, 2.2, 2.3, None, 4.4], np.float32, -1.1, 4.4, 1, 4),
-        ([-1.1, 2.2, 2.3, None, 4.4], np.float64, -1.1, 4.4, 1, 4),
+        ([1, 2, 2, None, 4], np.uint8, 'INT64', 1, 4, 1, 4, 0),
+        ([1, 2, 2, None, 4], np.uint16, 'INT64', 1, 4, 1, 4, 0),
+        ([1, 2, 2, None, 4], np.uint32, 'INT64', 1, 4, 1, 4, 0),
+        ([1, 2, 2, None, 4], np.uint64, 'INT64', 1, 4, 1, 4, 0),
+        ([-1, 2, 2, None, 4], np.int16, 'INT64', -1, 4, 1, 4, 0),
+        ([-1, 2, 2, None, 4], np.int32, 'INT64', -1, 4, 1, 4, 0),
+        ([-1, 2, 2, None, 4], np.int64, 'INT64', -1, 4, 1, 4, 0),
+        ([-1.1, 2.2, 2.3, None, 4.4], np.float32, 'FLOAT', -1.1, 4.4, 1, 4, 0),
```

Comment:
The current implementation returns with zero `distinct_count` and `INT64` for all of the integer types. 
I'm not sure about the correctness of these values, can someone confirm?

Reply:
This surprises me a bit. I would have expected that the small int types would have `INT32` as their physical type.

Reply:
No, `INT64` is not the expected. `distinct_count` was not computed, so I would expect `np.nan` or `None` as the result.

Reply:
I guess I need to dig deeper than to find why the physical_types are incorrect.

Reply:
How do I trigger `distinct_count` computation? 
Actually I can't find any [parquet-cpp tests](https://github.com/apache/parquet-cpp/blob/9586719acec86f790c00ac8797060b01a4f7de02/src/parquet/statistics-test.cc) mentioning `distinct_count`.